teps/0100-embedded-taskruns-and-runs-status-in-pipelineruns.md

```diff
@@ -171,12 +201,503 @@ In addition to grouping related resources, Tekton Results provides long term sto
+difficult for tools and users to parse for specific type of resource when looking at the
+status (compared to separate fields [as proposed](#1-add-minimal-embedded-status).
+
+### Add Minimal Embedded Status - Use Map
+
+Use maps for the new fields, as we do with the existing fields:
+
+```go
+type PipelineRunStatusFields struct {
+	...
+	TaskRuns map[string]*PipelineRunTaskRunStatus `json:"taskRuns,omitempty"`
+	TaskRunsStatuses map[string]*PipelineRunTaskRunMinimalStatus `json:"taskRunsStatuses,omitempty"`
+	Runs map[string]*PipelineRunRunStatus `json:"runs,omitempty"`
+	RunsStatuses map[string]*PipelineRunRunMinimalStatus `json:"runsStatuses,omitempty"`
+	...
+}
+```
+
+Taking the [example above](#example), this would be the `PipelineRun` status:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  ...
+spec:
+    ...
+status:
+    completionTime: "2020-05-04T02:19:14Z"
+    conditions:
+      - lastTransitionTime: "2020-05-04T02:19:14Z"
+        message: "Tasks Completed: 4, Skipped: 0"
+        reason: Succeeded
+        status: "True"
+        type: Succeeded
+    startTime: "2020-05-04T02:00:11Z"
+    taskRunStatus:
+      triggers-release-nightly-build:
+        conditions:
+          - lastTransitionTime: "2020-05-04T02:10:49Z"
+            message: All Steps have completed executing
+            reason: Succeeded
+            status: "True"
+            type: Succeeded
+```
+
+#### Discussion
+
+While this approach is consistent with existing code, it does not comply with the
+[Kubernetes API conventions][subobjects] that recommend against maps.
+The [main problem][why-subobjects] with maps is:
+> The crux of maps is that it isn't clear to the user what "left-hand side strings"
+> are "magic keywords" in the config system/API vs. which are user data.
+
+### Beta API - Use Booleans
+
+We could add a behavior flag - `enable-full-embedded-status` - used to configure
+whether `PipelineRuns` should contain the embedded status or minimal status of
+its `TaskRuns` and `Runs`.
+
+Following our [policy][behavior-flags] on updating behavior flags:
+* The `enable-full-embedded-status` flag will be *true* by default.
+* After 9 months in v1beta1, the `enable-full-embedded-status` flag will be flipped
+  to *false* by default.
+* As soon as the next release in v1beta1, the `enable-full-embedded-status` flag will
+  be removed as well as the full embedded status fields.
+
+#### Discussion
+
+While the behavior flag taking booleans solves for the options we need, the 
+[Kubernetes API conventions][primitives] warn "think twice about boolean fields"
+because "many ideas start as boolean but eventually trend towards a small set of
+mutually exclusive options". In the small chance that users want to have both full
+and minimal embedded statuses as they make transitions, using a boolean for this
+field would be limiting. Therefore, we prefer the alternative to using string aliases.
```

Comment:
Should this be the proposal directly? I think it makes sense, and there is no significant downside is using string compared to booleans... the only disadvantage could be ease of configuration for users, but documenting the valid values in the config map and docs should be enough to alleviate / solve this.